File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0 :
            return 0 

        if len(nums) == 1 :
            return 1 
            
        hashSet = set()

        for i in nums : 
            hashSet.add(i)

        res = 1
        longestConsecutive = 1
        for i in nums :
            if i + 1 in hashSet :
                temp = i 
                while True : 
                    if temp + 1 in hashSet :
                        longestConsecutive += 1
                        temp += 1 
                        continue 
                    res = max(res, longestConsecutive)
                    longestConsecutive = 1
                    break 

        return res
            

File: Data Structures & Algorithms/binary-search/submission-5.java
class Solution {
    public int recBinarySearch (int l, int r, int m, int[] nums, int target) {
        if (l > r) return - 1; 

        else if (nums[m] == target) return m; 

        else if (target < nums[m]) {
            r = m - 1;
            return recBinarySearch (l, r, ((l + r) / 2), nums, target); 
        }

        else {
            l = m + 1;
            return recBinarySearch (l, r, (l + r) / 2, nums, target); 
        }

    }
    
    public int search(int[] nums, int target) {
        return recBinarySearch(0, nums.length - 1, (nums.length - 1) / 2, nums, target);
    }
}
